### PR TITLE
Upgrade third-party websockets Python package to version 12.0

### PR DESCRIPTION
**Pull request by whimboo**
Currently we are running version 10.3 which is kinda outdated. With the upgrade some known failures might be fixed.

### CODE REVIEW

**Comment by whimboo** [comment]
Azure pipelines are still running with the minimal supported Python version 3.7 which is no longer officially supported. Gecko CI seems to no longer depend on it and has been upgraded to 3.8. If that is verified we should open a RFC (similar to the [bump to 3.7](https://github.com/web-platform-tests/rfcs/blob/master/rfcs/python_37.md)) to bump the minimum version to 3.8. For now I've pushed such a commit here on this stack.

**Comment by whimboo** [comment]
I've created a RFC for web-platform tests to bump the minimum Python version from 3.7 to 3.8: https://github.com/web-platform-tests/rfcs/pull/189

Due to the complexity of work I'll move out the version bump to a different PR once the RFC got accepted.

**Comment by whimboo** [comment]
> I've created a RFC for web-platform tests to bump the minimum Python version from 3.7 to 3.8: [web-platform-tests/rfcs#189](https://github.com/web-platform-tests/rfcs/pull/189)

The version bump from 3.7 to 3.8 is handled on https://github.com/web-platform-tests/wpt/pull/45575.

File: webdriver/tests/support/fixtures.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 PAGE_LOAD_TIMEOUT = 3
 IMPLICIT_WAIT_TIMEOUT = 0
 
-# The webdriver session can outlive a pytest session
+# The webdriver session can outlive a pytest session.
```
Comment (whimboo) [inline webdriver/tests/support/fixtures.py:22]: Note that this change will be reverted before landing. It was used to trigger all the wdspec tests which are the only consumers of websockets in wpt.

**Comment by whimboo** [comment]
Tests for Chrome and Firefox do not show any particular failure related to the websockets package with this upgrade applied. Chrome-dev-stability fails because of a lot of failures of type `Failed to create Chrome process` but there is no stack. @nechaev-chromium or @sadym-chromium are you aware of issues those days?

**Comment by nechaev-chromium** [comment]
I have no immediate explanation for this. Going to check this locally.

**Comment by nechaev-chromium** [comment]
It does not reproduce locally. Is it possible to see chromedriver.log (--webdriver-arg="--log-path=$chromedriver_log_path")? I cannot find it in the artifacts.

**Comment by whimboo** [comment]
> It does not reproduce locally. Is it possible to see chromedriver.log (--webdriver-arg="--log-path=$chromedriver_log_path")? I cannot find it in the artifacts.

Yeah, I cannot reproduce locally as well. But I don't know where such an option would have to be set and if we maybe should store driver logs by default as artifacts. @jgraham what do you think?

@nechaev-chromium but I think that this problem is not related to the bump of this 3rd party Python package. Do you agree with it?

**Review by jgraham — APPROVED** [review]

**Pull request merged**